x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/TestCaseSupplier.java

```diff
@@ -1098,51 +1098,57 @@ public static class TestCase {
         /**
          * The {@link Source} this test case should be run with
          */
-        private Source source;
+        private final Source source;
```

Comment:
I've modified this to copy the test case when you call the modification methods. That's a bit more normal for java stuff.